wisp/wisp-resource-loader/src/main/kotlin/wisp/resources/ClasspathResourceLoaderBackend.kt

```diff
@@ -79,10 +79,11 @@ object ClasspathResourceLoaderBackend : ResourceLoader.Backend() {
                 val childFilePath = File(entry.name).toPath().normalize()
                 if (!childFilePath.startsWith(Paths.get(pathPrefix))) continue
 
-                var endIndex = entry.name.indexOf("/", pathPrefix.length)
-                if (endIndex == -1) endIndex = entry.name.length
+                val childFilePathStr = childFilePath.toString()
+                var endIndex = childFilePathStr.indexOf("/", pathPrefix.length)
+                if (endIndex == -1) endIndex = childFilePathStr.length
 
-                result += entry.name.substring(pathPrefix.length, endIndex)
+                result += childFilePathStr.substring(pathPrefix.length, endIndex)
```

Comment:
this is fine but it might be better to do instead of string manipulation:
```
val prefix = Paths.get(pathPrefix) // this can be extracted from line 80
if (childFilePath.getNameCount() > prefix.getNameCount()) {
  result += childFilePath.name(prefix.getNameCount()).toString()
}
```

Reply:
This looks so much cleaner! Updated the code.